tools/introspection/include/iceoryx_introspection/introspection_app.hpp

```diff
@@ -62,7 +62,6 @@ static const std::map<PrettyOptions, uint32_t> prettyMap = {
 
 
 /// @brief base class for introspection
-/// @todo iox-#1692 Wrap all C calls with posixCall
```

Comment:
I think this comment here meant to wrap all C operating system calls e.g. those in `initTerminal`. There might be others.

```
    initscr();
    curs_set(0);
    start_color();
    init_pair(static_cast<uint8_t>(ColorPairs::redOnBlack), COLOR_RED, COLOR_BLACK);
    init_pair(static_cast<uint8_t>(ColorPairs::whiteOnRed), COLOR_WHITE, COLOR_RED);
```

@elfenpiff Can you confirm?

Reply:
@elfenpiff Can you confirm if my understanding to the use of `posixCall` is right ? or I need to wrap all calls from c written libraries like `ncurses` ?

cc. @mossmaurice

Reply:
@zmostafa You are completely right with your understanding. The posix call function is for posix c function calls only. It requires that the either the return value is the errno or the return value indicates an errno and the errno is set.

C functions which can never fail (even when they are posix function) do not need to be wrapped.

The intention of the posixCall was, that the error handling and all the side conditions are hard to handle and it is easy to miss something. The posixCall does some rudimentary error handling already for you, resets the errno = 0 before calling the posix function and repeats the call if it has failed and the errno is set to EINTR. Something which you definitely do not want from an ncurses function.

Reply:
@elfenpiff Thanks for the detailed explanation.

So, my understanding is like:
 1. `posixCall` can be used with non-posix calls, to use its error handling features.
 2. I should look on `ncurses` methods and their returned error number to wrap them in a `posixCall`

Does this make sense now ?

Reply:
> The posixCall does some rudimentary error handling already for you, resets the errno = 0 before calling the posix function and **repeats the call if it has failed and the errno is set to EINTR. Something which you definitely do not want from an ncurses function.**

@zmostafa From my previous comment. This is why I would not use the `posixCall` to wrap non-posix calls like for instance ncurses functions.

Reply:
@elfenpiff my bad, I thought you do not want a failure from ncureses, and re-trying make sense here :sweat_smile: Thanks again for your explanation.

I wrapped `getchar()` already and I think nothing needs to be done more here,unless I missed a concrete posix call here :eyes:

Reply:
@zmostafa No problem at all! I think you finished them all and now lets get it merged.